service-clients-gen/src/main/java/com/flipkart/poseidon/serviceclients/generator/ServiceGenerator.java

```diff
@@ -403,14 +404,15 @@ private JInvocation createRequest(ServiceIDL serviceIdl, JCodeModel jCodeModel,
             }
         }
 
-        if (endPoint.isIncludeMetaInfo()) {
+        if (endPoint.isIncludeMetaInfo() && endPoint.getMetaInfoMeta() != null && endPoint.getMetaInfoMeta().isDynamicCommandName()) {
```

Comment:
divide this into two conditions, one for endPoint.isIncludeMetaInfo() and the other for the rest. Will make it easier to add other features